src/data.rs

```diff
@@ -20,16 +21,19 @@ pub enum Struct<T> {
 #[derive(Debug, Clone)]
 pub struct UnitStruct {
     pub(crate) private: (),
```

Comment:
You can safely remove the private field now that you have added the attrs field. The private field was only there to prevent people from instantiating a UnitStruct themselves.

Reply:
Let's do that in different PR, since the scope of #25 (this PR) is already quite big.

Reply:
That's fine with me.